index.bs

```diff
@@ -491,10 +537,13 @@ When the <dfn method for="XRSession">cancelAnimationFrame(|handle|)</dfn> method
 
 <div class="algorithm" data-algorithm="run-animation-frames">
 
-  When an {{XRSession}} |session| receives updated [=viewer=] state from the [=XR device=], it runs an <dfn>XR animation frame</dfn> with a timestamp |now| and an {{XRFrame}} |frame|, which MUST run the following steps:
+When an {{XRSession}} |session| receives updated [=viewer=] state from the [=XR device=], it runs an <dfn>XR animation frame</dfn> with a timestamp |now| and an {{XRFrame}} |frame|, which MUST run the following steps:
 
   1. Let |callbacks| be a list of the entries in |session|'s [=list of animation frame callback=], in the order in which they were added to the list.
   1. Set |session|'s [=list of animation frame callbacks=] to the empty list.
+  1. Let |newRenderState| be a copy of |session|'s [=pending render state=].
+  1. Let |renderStatePromises| be a list of the entries in |session|'s [=list of pending render state promises=], in the order in which they were added to the list.
+  1. Set |session|'s [=list of pending render state promises=] to the empty list.
```

Comment:
Does setting the session's list of pending render state promises to an empty list mean we resolve the promise at this step?  If not, when do we resolve the promise?

How does the ordering of the promise resolution interleave with the RAF callbacks?  Could we end up in a state where the `updateRenderState` promise resolves after the next frame's Session.RAF callback?  If so, this seems unnatural as the web developer would likely expect that the "render state accepted" promise would resolve BEFORE they receive the RAF callback with the render state applied.  

What would be the downside of dispensing with the promise altogether and have `updateRenderState` overwrite the pending render state when called?  Pending render state becomes active render state right before RAF callbacks are serviced.

Reply:
> Does setting the session's list of pending render state promises to an empty list mean we resolve the promise at this step? If not, when do we resolve the promise?

I second this question from Rafael :)

Reply:
>What would be the downside of dispensing with the promise altogether and have updateRenderState overwrite the pending render state when called? Pending render state becomes active render state right before RAF callbacks are serviced.

The purpose of the promise is just to let developers know when the changes have been applied.  It's entirely possible no one will use the promise returned, but I'm not certain. That said, I think Brandon's latest changes mostly address the rest of your point.

Reply:
I understand that the purpose is to let developers know when the changes have been applied.  However, as the spec is currently written, the developer would receive the promise resolution after the RAF callback with the new values has been fired.  This is true even in the latest revision.

Put another way, a naïve developer would call updateRenderState and then render with the old values until they received a promise resolution signifying the 'all clear'. With the current spec language, they would end up rendering one or more frames with the wrong values.